src/util/ImageUploader.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+class QJsonObject;
+
+namespace chatterino {
+
+class Settings;
+
+namespace imageuploader {
+namespace detail {
+
+QJsonObject exportSettings(Settings &s);
+bool importSettings(const QJsonObject &settingsObj, Settings &s);
+bool validateImportJson(const QString &clipboardText, QJsonObject &settingsObj);
+void parseAndApplyHeaders(const QJsonObject &headersObj, Settings &s);
```

Comment:
This function is only ever used by the internal pieces - you can make it a source-only function.
Additionally, it would be nice if it only did the parsing, letting the `importSettings` function do the importing.
I imagine the signature would change to something like
```c++
QStringList parseHeaders(const QJsonObject &headersObj);
```